src/jit/alloc.h

```diff
@@ -51,20 +47,25 @@ struct ArenaAllocator
     BYTE* m_nextFreeByte; 
     BYTE* m_lastFreeByte;
 
-    IEEMemoryManager* m_memoryManager;
+    bool isInitialized();
 
-    void* allocateNewPage(size_t size);
-
-    // The following methods are used for mark/release operation.
-    void mark(MarkDescriptor& mark);
-    void reset(MarkDescriptor& mark);
+    void* allocateNewPage(size_t size, bool canThrow);
 
     void* allocateHostMemory(size_t size);
     void freeHostMemory(void* block);
 
 public:
-    bool initialize(IEEMemoryManager* memoryManager, bool shouldPreallocate);
-    void destroy();
+    ArenaAllocator();
+    ArenaAllocator(IEEMemoryManager* memoryManager);
+    ArenaAllocator& operator=(ArenaAllocator&& other);
```

Comment:
The move-ctor could be protected, since you only expect the PooledAllocator to call it.